i18n/check_and_integrate_translations.py

```diff
@@ -0,0 +1,180 @@
+import os
+import sys
+import argparse
+import subprocess
+import shutil
+import glob
+from match_and_merge_po_translations import parse_po_file
+from dataclasses import dataclass
+
+
+@dataclass
+class TranslationsData:
+    language_code: str
+    directory_path: str
+    total_strings: int = 0
+    missing_translations: int = 0
+    fuzzy_translations: int = 0
+
+    _completion_rate: int = -1
+
+    def get_completion_rate(self) -> int:
+        """Returns the completion rate of the language with an int between 0 and 100, with 100 representing 100% completion."""
+        if self._completion_rate == -1:
+            todo_count = self.missing_translations + self.fuzzy_translations
+            self._completion_rate = 100 - (100 * todo_count // self.total_strings)
+
+        return self._completion_rate
+
+
+@dataclass
+class Args:
+    translations_path: str
+    threshold: int
+    skip_extract: bool
+    skip_sync: bool
```

Comment:
This is another change I wanted to highlight. It's personal taste, but I wanted to show this in case you've never used dataclasses - they came in python 3.9 or something. They allow you to quickly make data structures basically, mainly to benefit from better code completion.

Again, personal taste, I just like to get help from the code linter when making typos or doing incorrect assignments (you don't get errors for this with dictionaries).

Reply:
From what I understood it is basically a class with automated getters and setters ? 
Well it is indeed a better solution for a complex data structure.